### PR TITLE
Generate stubs for Automated V&V Phase 1

### PR DESCRIPTION
**Pull request by patricktnast**
### Description

- *Category*: 
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5916

### Changes and notes
 
I stubbed out the classes from phase 1 in accordance with the design document
https://hub.ihme.washington.edu/spaces/SSE/pages/381354690/Phase+1+Standard+Metric+Comparisons

I left a few signatures blank, as well as the docstrings; I didn't want to fill out too much too quickly as things are subject to change.

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_testing_utils/automated_validation/interface.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+class ValidationContext:
+    def __init__(self, results_dir: str | Path, age_groups: pd.DataFrame | None):
+        self.data_loader = DataLoader(results_dir)
+        self.comparisons = LayeredConfigTree()
+
+    def sim_outputs(self):
+        return self.data_loader.sim_outputs()
+
+    def artifact_keys(self):
```
Comment (rmudambi) [inline src/vivarium_testing_utils/automated_validation/interface.py:19]: Nit: I like methods to be verbs and attributes/properties to be nouns. We can definitely discuss if we think that ought to be a rule.

File: src/vivarium_testing_utils/automated_validation/interface.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        )
+
+    def verify(self, comparison_key: str, stratifications: list[str] = []):
+        self.comparisons[comparison_key].verify(stratifications)
+
+    def plot_comparison(self, comparison_key: str, type: str, kwargs):
+        return plot_utils.plot_comparison(self.comparisons[comparison_key], type, kwargs)
+
+    def generate_comparisons(self):
+        pass
```
Comment (rmudambi) [inline src/vivarium_testing_utils/automated_validation/interface.py:38]: Nit: maybe raise `NotImplementedError` rather than passing?

File: src/vivarium_testing_utils/automated_validation/interface.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        test_data = self.data_loader.get_dataset(measure_key, test_source)
+        ref_data = self.data_loader.get_dataset(measure_key, ref_source)
+        self.comparisons.update(
+            [measure_key], Comparison(measure_key, test_data, ref_data, stratifications)
+        )
+
+    def verify(self, comparison_key: str, stratifications: list[str] = []):
+        self.comparisons[comparison_key].verify(stratifications)
+
+    def plot_comparison(self, comparison_key: str, type: str, kwargs):
```
Comment (rmudambi) [inline src/vivarium_testing_utils/automated_validation/interface.py:34]:
```suggestion
    def plot_comparison(self, comparison_key: str, type: str, **kwargs):
```

**Review by rmudambi — APPROVED** [review]
Looks good. I have a few nits that we can talk about if you'd like
